doc/source/dev/howto_build_docs.rst

```diff
@@ -55,6 +55,18 @@ Doxygen_ can be installed with::
 
     pip install -r requirements/doc_requirements.txt
 
+.. note::
+
+    Since NumPy 2.0 is still in development, so we need to install
+    nightly build packages for dependency. The command is as below::
+
+        pip install --pre --force-reinstall --extra-index-url \
+        https://pypi.anaconda.org/scientific-python-nightly-wheels/simple \
+        -r requirements/doc_requirements.txt
+
+    For MacOS Arm platform, the command above may not install needed packages
+    so NumPy docs can't be built.
+
```

Comment:
While true, I'd vote to leave this out of the docs. Hopefully it will be fixed in the very near future and would be one less thing to keep track of.